capstone/capapi/views.py

```diff
@@ -75,64 +76,38 @@ class CaseViewSet(BaseViewMixin, mixins.RetrieveModelMixin, mixins.ListModelMixi
-            except models.CaseMetadata.DoesNotExist as e:
-                return JsonResponse({
-                    'message': 'Unable to find case with matching slug: %s' % e
-                }, status=404, )
-
+    def get_serializer_class(self, *args, **kwargs):
+        full_case = self.request.query_params.get('full_case', 'false').lower()
+        if full_case == 'true':
+            return serializers.CaseSerializerWithCasebody
         else:
-            cases = self.queryset.all()
-            for backend in list(self.filter_backends):
-                cases = backend().filter_queryset(self.request, cases, self)
-
-            # user is requesting a zip but there is nothing to zip, so 404 is the right response.
-            # See https://stackoverflow.com/a/11760249/307769
-            if not cases.exists():
-                return JsonResponse({
-                    'message': 'Request did not return any results.',
-                }, status=404, )
-
-            case_list = self.paginate_queryset(cases)
-
-        blacklisted_case_count = len(list((case for case in case_list if not case.jurisdiction.whitelisted)))
-
-        try:
-            # check user's case allowance against blacklisted
-            user = serializers.UserSerializer().verify_case_allowance(self.request.user, blacklisted_case_count)
-        except ValidationError as err:
-            return JsonResponse(err.detail, status=403)
-
-        filename = resources.create_zip_filename(case_list)
-
-        case_response = serializers.CaseSerializerWithCasebody(case_list, many=True, context={'request': self.request})
-        case_data = case_response.data
-        response = resources.create_download_response(filename=filename, content=case_data)
-        user.update_case_allowance(case_count=blacklisted_case_count)
-
-        return response
+            return self.serializer_class
 
     def list(self, *args, **kwargs):
-        if not self.request.query_params.get('type') == 'download':
-            return super(CaseViewSet, self).list(*args, **kwargs)
-        else:
-            return self.download(**kwargs)
+        self.serializer_class = self.get_serializer_class(self, *args, **kwargs)
+        cite = self.request.query_params.get('cite', None)
+        if cite:
+            self.queryset = self.queryset.filter(citation__normalized_cite=cite)
+        return super(CaseViewSet, self).list(*args, **kwargs)
 
     def retrieve(self, *args, **kwargs):
-        if not self.request.query_params.get('type') == 'download':
-            return super(CaseViewSet, self).retrieve(*args, **kwargs)
+        slugified = slugify(kwargs[self.lookup_field])
+        # for user's convenience, if user gets /cases/case-citation or /cases/Case Citation
+        # we redirect to /cases/?cite=case-station
+        if '-' in slugified:
+            query_params = ''.join(list(map(lambda k: '&%s=%s' % (k, self.request.query_params[k]), self.request.query_params)))
```

Comment:
This won't work if values have `&` or `=` in them -- you want to encode the parameters. It's also a good idea to use the reverse() function for the URL. Something like this should work (borrowed from Perma's current_query_string templatetag):

```
import urllib
from rest_framework.reverse import reverse

query_string = urllib.urlencode(dict(self.request.query_params, cite=slugified), doseq=True)
new_url = reverse('api:cases-list', request=self.request) + "?" + query_string
```